task/transcode.go

```diff
@@ -115,6 +116,24 @@ func fileWriter(size int64) WriteSeekCloser {
 	}
 }
 
+type SegmentCounter struct {
+	size  uint64
+	count uint64
```

Comment:
**nit:** Would also avoid `count` here not to seem like it's counting the number of segments. A suggestion:
```suggestion
	totalSize  uint64
	readSize uint64
```
(or can even drop the `Size` suffix if the struct name is clear enough)